src/ArcxAnalyticsSdk.ts

```diff
@@ -184,13 +185,17 @@ export class ArcxAnalyticsSdk {
 
   private async _getCurrentChainId(): Promise<string> {
     if (!window.ethereum) {
-      throw new Error('ArcxAnalyticsSdk::_getCurrentChainId: No ethereum provider found')
+      const errorMsg = 'ArcxAnalyticsSdk::_getCurrentChainId: No ethereum provider found'
+      this._reportError(errorMsg)
+      throw new Error(errorMsg)
```

Comment:
I'd make a pure utility function `reportError(sdk:ArcxAnalyticsSdk, error: any)` that posts the error then throws. This way we can use it everywhere in any file and save some additional lines

Reply:
`reportError` is private so i extracted `_reportErrorAndThrow`